src/fides/api/models/sql_models.py

```diff
@@ -514,6 +532,124 @@ def create(
+            else_=None,
+        )
+
+    @hybrid_property
+    async def _publisher_override_legal_basis_join(self) -> Optional[str]:
+        """Returns the instance-level overridden required legal basis"""
+        query: Select = select([TCFPublisherOverride.required_legal_basis]).where(
+            TCFPublisherOverride.purpose == self.purpose
+        )
+        async_session: AsyncSession = async_object_session(self)
+        async with async_session.begin():
+            result = await async_session.execute(query)
+            return result.scalars().first()
+
+    @_publisher_override_legal_basis_join.expression
+    def _publisher_override_legal_basis_join(cls) -> ScalarSelect:
+        """Returns the class-level overridden required legal basis"""
+        return (
+            select([TCFPublisherOverride.required_legal_basis])
+            .where(TCFPublisherOverride.purpose == cls.purpose)
+            .as_scalar()
+        )
+
+    @hybrid_property
+    async def _publisher_override_is_included_join(self) -> Optional[bool]:
+        """Returns the instance-level indication of whether the purpose should be included"""
+        query: Select = select([TCFPublisherOverride.is_included]).where(
+            TCFPublisherOverride.purpose == self.purpose
+        )
+        async_session: AsyncSession = async_object_session(self)
+        async with async_session.begin():
+            result = await async_session.execute(query)
+            return result.scalars().first()
+
+    @_publisher_override_is_included_join.expression
+    def _publisher_override_is_included_join(cls) -> ScalarSelect:
+        """Returns the class-level indication of whether the purpose should be included"""
+        return (
+            select([TCFPublisherOverride.is_included])
+            .where(TCFPublisherOverride.purpose == cls.purpose)
+            .as_scalar()
+        )
+
+    @hybrid_property
+    async def overridden_legal_basis_for_processing(self) -> Optional[str]:
+        """
+        Instance-level override of the legal basis for processing based on
+        publisher preferences.
+        """
+        if not (
+            CONFIG.consent.override_vendor_purposes
+            and self.flexible_legal_basis_for_processing
+        ):
+            return self.legal_basis_for_processing
+
+        is_included: Optional[bool] = await self._publisher_override_is_included_join
+
+        if is_included is False:
+            # Overriding to False to match behavior of class-level override.
+            # Class-level override of legal basis to None removes Privacy Declaration
+            # from Experience
+            return None
+
+        overridden_legal_basis: Optional[str] = (
+            await self._publisher_override_legal_basis_join
+        )
+
+        return (
+            overridden_legal_basis
+            if overridden_legal_basis  # pylint: disable=using-constant-test
+            else self.legal_basis_for_processing
+        )
```

Comment:
This is the matching instance-level hybrid property that is used for supplementing the system-level response so we can return the overridden legal basis via API. 

Some things I don't like here, since I need to join with the publisher overrides, I need a session, and I only have the async session available to me in the GET* /system contexts.    When FastAPI serializes the response, you get issues with coroutines not being awaited, so instead, I am adding a property of a different name to the PrivacyDeclarationResponse and just stashing it on the PrivacyDeclaration at runtime myself before the response is serialized.